website/addons/box/model.py

```diff
@@ -303,94 +189,11 @@ def create_waterbutler_log(self, auth, action, metadata):
-            name = removed.fullname
-            return (
-                u'The Box add-on for this {category} is authenticated by {name}. '
-                'Removing this user will also remove write access to Box '
-                'unless another contributor re-authenticates the add-on.'
-            ).format(**locals())
-
-    # backwards compatibility
-    before_remove_contributor = before_remove_contributor_message
-
-    def after_fork(self, node, fork, user, save=True):
-        """After forking, copy user settings if the user is the one who authorized
-        the addon.
-
-        :return: A tuple of the form (cloned_settings, message)
-        """
-        clone, _ = super(BoxNodeSettings, self).after_fork(
-            node=node, fork=fork, user=user, save=False
-        )
-
-        if self.user_settings and self.user_settings.owner == user:
-            clone.user_settings = self.user_settings
-            message = (
-                'Box authorization copied to forked {cat}.'
-            ).format(cat=fork.project_or_component)
+    def after_delete(self, node=None, user=None):
+        if user:
+            self.deauthorize(Auth(user=user), add_log=True)
         else:
-            message = (
-                u'Box authorization not copied to forked {cat}. You may '
-                'authorize this fork on the <u><a href="{url}">Settings</a></u> '
-                'page.'
-            ).format(
-                url=fork.web_url_for('node_setting'),
-                cat=fork.project_or_component
-            )
-        if save:
-            clone.save()
-        return clone, message
-
-    def after_remove_contributor(self, node, removed, auth=None):
-        """If the removed contributor was the user who authorized the Box
-        addon, remove the auth credentials from this node.
-        Return the message text that will be displayed to the user.
-        """
-
-        if self.user_settings and self.user_settings.owner == removed:
-
-            # Delete OAuth tokens
-            self.user_settings = None
-            self.save()
-            message = (
-                u'Because the Box add-on for {category} "{title}" was authenticated '
-                u'by {user}, authentication information has been deleted.'
-            ).format(
-                category=node.category_display,
-                title=node.title,
-                user=removed.fullname
-            )
-
-            if not auth or auth.user != removed:
-                url = node.web_url_for('node_setting')
-                message += (
-                    u' You can re-authenticate on the <u><a href="{url}">Settings</a></u> page.'
-                ).format(url=url)
-            #
-            return message
-
-    def after_delete(self, node, user):
-        self.deauthorize(Auth(user=user), add_log=True)
+            self.deauthorize(add_log=True)
         self.save()
+
+    on_delete = after_delete
```

Comment:
It's a super-super-superclass [override](https://github.com/mfraezz/osf.io/blob/feature/box-oauth-refactor/website/addons/base/__init__.py#L233-L235), used when [deleting](https://github.com/mfraezz/osf.io/blob/feature/box-oauth-refactor/website/addons/base/__init__.py#L209) nodes.